### DISCUSSION TITLE
Static filter rule syntax

### DISCUSSION THREAD

**Discussion by SJinfinite** [uBlockOrigin/uAssets#18631] [Q&A] [answered]
### Prerequisites

- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] I am running the latest version of uBlock Origin (uBO).

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

> 静态过滤规则中,类似

**In static filtration rules, similar:**
```adb
www.huya.com###diy-activity-icon-1
www.huya.com###diy-activity-icon-2
www.huya.com###diy-activity-icon-3
www.huya.com###diy-activity-icon-4
www.huya.com###diy-activity-icon-...
```
> 是否有通配符或语法可以表示出任意数字的情况,
类似的,还有

**Whether there is a compatriot or grammar can indicate the situation of any number,
similarly, there are:**
```adb
search.bilibili.com##div.mb_x40.col_xl_1_7.col_md_2.col_xs_1_5.col_3:nth-of-type(1)
search.bilibili.com##div.mb_x40.col_xl_1_7.col_md_2.col_xs_1_5.col_3:nth-of-type(2)
search.bilibili.com##div.mb_x40.col_xl_1_7.col_md_2.col_xs_1_5.col_3:nth-of-type(3)
search.bilibili.com##div.mb_x40.col_xl_1_7.col_md_2.col_xs_1_5.col_3:nth-of-type(...)
```
> 本人尝试过-*的表示方式,结果是对应网站的全部过滤规则都失效了,希望能得到帮助.

**I have tried the way of-*, and the result is that all the filter rules of the corresponding website have failed. I hope to get help.**

### URL(s) where the issue occurs.

```
www.huya.com
search.bilibili.com
```

### Configuration

<details>

```yaml
uBlock Origin 1.49.3b10:
 since: 37d.2h.42m
Chromium: 95
filterset (summary):
 network: 109732
 cosmetic: 49707
 scriptlet: 17818
 html: 0
listset (total-discarded, last-updated):
 default:
  user-filters: 268-0, never
  CHN-0: 26223-117, 1h.28m
  easylist: 67023-110, 1d.8h.58m
  easyprivacy: 30602-71, 1d.8h.57m
  plowe-0: 3735-954, 8d.1h.55m
  ublock-badware: 7516-137, 3d.1h.7m
  ublock-filters: 33560-158, 3d.1h.6m
  ublock-privacy: 396-2, 6d.5h.54m
  ublock-quick-fixes: 253-14, 4h.46m
  ublock-unbreak: 2049-35, 3d.1h.4m
  urlhaus-1: 7381-1, 2h.37m
filterset (user): [array of 268 redacted]
userSettings:
 advancedUserEnabled: true
 cloudStorageEnabled: true
 suspendUntilListsAreLoaded: true
hiddenSettings: [none]
supportStats:
 allReadyAfter: 401 ms
 maxAssetCacheWait: 104 ms
```
</details>

**Comment by MasterKia** [comment, marked as answer]
Try:
```adb
huya.com##[id*="diy-activity-icon-"]

search.bilibili.com##div.mb_x40.col_xl_1_7.col_md_2.col_xs_1_5.col_3
```

***
> uBlock Origin 1.49.3b10:
Chromium: 95

Update uBO and your browser, they're old.

***

> - [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).

If you have further questions, please ask in Reddit.

**Reply by SJinfinite** [reply]
十分感谢您的帮助,我的问题已经解决!

**Comment by gorhill** [comment]
huya.com##[id^="diy-activity-icon-"]
    search.bilibili.com##div.mb_x40.col_xl_1_7.col_md_2.col_xs_1_5.col_3

For the second case, use the right-hand slider in the element picker to lower specificity such that the element picker will drop the `:nth-of-type(...)` part. See documentation: https://github.com/gorhill/uBlock/wiki/Element-picker

**Reply by SJinfinite** [reply]
十分感谢您的帮助,我的问题已经解决!
